internal/controlplane/filemgr/config.go

```diff
@@ -0,0 +1,33 @@
+package filemgr
+
+import (
+	"os"
+	"path/filepath"
+)
+
+type config struct {
+	cacheDir string
+}
+
+// An Option updates the config.
+type Option = func(*config)
+
+// WithCacheDir returns an Option that sets the cache dir.
+func WithCacheDir(cacheDir string) Option {
+	return func(cfg *config) {
+		cfg.cacheDir = cacheDir
+	}
+}
+
+func newConfig(options ...Option) *config {
+	cfg := new(config)
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		cacheDir = filepath.Join(os.TempDir())
```

Comment:
If `os.UserCacheDir` fails, I believe it would be safer to create a more random subdirectory (eg, ioutil.TempDir) and ensure the permissions are properly restricted before using it.  Thoughts?

Reply:
I added a uuid and updated the permissions to be 700.